### PR TITLE
License policy plugin

### PR DESCRIPTION
**Pull request by pombredanne**
This is the PR for #214

### CODE REVIEW

**Comment by pombredanne** [comment]
@majurg the test failure is because some path is reported as absolute rather than relative.

**Comment by majurg** [comment]
@pombredanne How do I prevent the plugin system from using absolute path?

File: src/scancode/plugin_license_policy.py [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+    """
+
+    attributes = dict(license_policy=attr.ib(default=attr.Factory(dict)))
+
+    sort_order = 9
+
+    options = [
+        CommandLineOption(('--license-policy',),
+            multiple=False,
+            metavar='<license_policy>',
```
Comment (pombredanne) [inline src/scancode/plugin_license_policy.py]: Just curious: are we using metavars this way elsewhere? or rather a plain `FILE` ?

File: src/scancode/plugin_license_policy.py [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+                resource_license_keys = [entry.get('key') for entry in resource.licenses]
+
+                for key in resource_license_keys:
+                    if key in license_policies.keys():
+                        # Apply the policy to the Resource
+                        resource.license_policy = license_policies[key]
+                        codebase.save_resource(resource)
+
+            except AttributeError:
+                # add license_policy regardless if there is --license info or not
```
Comment (pombredanne) [inline src/scancode/plugin_license_policy.py]: I was about to say that another way could have been to require a license scan... But this is actually better as this always works!

File: src/scancode/plugin_license_policy.py (hunk 1 above)
Comment (majurg) [reply src/scancode/plugin_license_policy.py]: Ok I will make that change

File: src/scancode/plugin_license_policy.py (hunk 2 above)
Comment (pombredanne) [reply src/scancode/plugin_license_policy.py]: I think we are could have this both as a separate plugin and we should be able to run on a "virtual" codebase alright as it is

**Comment by pombredanne** [comment]
@majurg the failure is based on @JonoYang pending fix in #1018 , so I am merging as is.
Thank you!

**Pull request merged**